source/AI.cpp

```diff
@@ -2137,6 +2137,17 @@ void AI::DoScatter(Ship &ship, Command &command)
 }
 
 
+const bool AI::ShouldActFrugally(const Ship &ship) const
+{
+	auto ait = allyStrength.find(ship.GetGovernment());
```

Comment:
You can remove these and simplify the logic of this method by calling AI::AllyStrength and AI::EnemyStrength, as those handle the iterators and give you the strength value you are seeking.

i.e. `if(AllyStrength < EnemyStrength || health check)` -> don't act frugally, otherwise act frugally.

Reply:
still have to do this